schema.sql

```diff
@@ -0,0 +1,88 @@
+DROP TABLE IF EXISTS application;
+DROP TYPE IF EXISTS application_status;
+DROP TABLE IF EXISTS applicant_phone_number;
+DROP TYPE IF EXISTS phone_type;
+DROP TABLE IF EXISTS applicant_address;
+DROP TABLE IF EXISTS applicant;
+DROP TABLE IF EXISTS advisor;
+DROP TYPE IF EXISTS role;
+DROP TABLE IF EXISTS "user";
+
+CREATE TABLE "user"
+(
+    id       BIGSERIAL,
+    email    TEXT NOT NULL,
+    username TEXT NOT NULL,
+    CONSTRAINT user_PK PRIMARY KEY (id),
+    CONSTRAINT user_email_UQ UNIQUE (email),
+    CONSTRAINT user_email_CH CHECK (email LIKE '%@%'),
+    CONSTRAINT user_username_UQ UNIQUE (username)
+);
+
+CREATE TYPE role AS ENUM ('associate', 'partner', 'senior');
+
+CREATE TABLE advisor
+(
+    id      BIGSERIAL,
+    user_id BIGINT NOT NULL,
+    role    role   NOT NULL,
+    CONSTRAINT advisor_PK PRIMARY KEY (id),
+    CONSTRAINT advisor_user_FK FOREIGN KEY (user_id) REFERENCES "user",
+    CONSTRAINT advisor_user_id_UQ UNIQUE (user_id)
+);
+
+CREATE TABLE applicant
+(
+    id         BIGSERIAL,
+    user_id    BIGINT NOT NULL,
+    first_name TEXT   NOT NULL,
+    last_name  TEXT   NOT NULL,
+    ssn        TEXT   NOT NULL,
+    CONSTRAINT applicant_PK PRIMARY KEY (id),
+    CONSTRAINT applicant_user_FK FOREIGN KEY (user_id) REFERENCES "user",
+    CONSTRAINT applicant_user_id_UQ UNIQUE (user_id)
+);
+
+CREATE TABLE applicant_address
+(
+    id     BIGINT,
+    city   TEXT NOT NULL,
+    street TEXT NOT NULL,
+    number INT  NOT NULL,
+    zip    INT  NOT NULL,
+    apt    INT  NOT NULL,
```

Comment:
If applicant lives in a house and there is no apt number ?

Reply:
Good thought. I'll add it

Reply:
Done